c/driver/flightsql/dremio_flightsql_test.cc

```diff
@@ -0,0 +1,83 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <adbc.h>
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest-matchers.h>
+#include <gtest/gtest-param-test.h>
+#include <gtest/gtest.h>
+#include <nanoarrow/nanoarrow.h>
+#include "validation/adbc_validation.h"
+#include "validation/adbc_validation_util.h"
+
+class DremioFlightSQLQuirks : public adbc_validation::DriverQuirks {
+ public:
+  AdbcStatusCode SetupDatabase(struct AdbcDatabase* database,
+                               struct AdbcError* error) const override {
+    const char* uri = std::getenv("ADBC_DREMIO_FLIGHTSQL_URI");
+    const char* user = std::getenv("ADBC_DREMIO_FLIGHTSQL_USER");
+    const char* pass = std::getenv("ADBC_DREMIO_FLIGHTSQL_PASS");
+    EXPECT_EQ(AdbcDatabaseSetOption(database, "uri", uri, error), ADBC_STATUS_OK);
```

Comment:
nit: you should be able to `ASSERT_THAT(..., IsOkStatus(&error));`

Reply:
Can't use `ASSERT_THAT` because of:

```bash
/home/zeroshade/Projects/GitHub/arrow-adbc/c/driver/flightsql/dremio_flightsql_test.cc:36:5: error: void value not ignored as it ought to be
   36 |     ASSERT_THAT(AdbcDatabaseSetOption(database, "password", pass, error), IsOkStatus(&error));
```

but i can switch it to `EXPECT_THAT` and use `IsOkStatus` so that works.